src/operator/batch_norm-inl.h

```diff
@@ -50,20 +52,294 @@ struct BatchNormParam : public dmlc::Parameter<BatchNormParam> {
   }
 };
 
-template<typename xpu>
-class BatchNormOp : public Operator {
+/*! \brief Batch normalization operator */
+template<typename xpu, typename Dtype, typename AccType>
+class BatchNormOp : public Operator
+                  , public Callbacker<Operator> {
+  typedef ::nnvm::TShape TShape;
+  typedef ::mxnet::TBlob TBlob;
```

Comment:
they get mixed up sometimes with mshadow::TShape/TBlob. They are there because of compile issues.